sys/mips/include/cheriasm.h

```diff
@@ -576,12 +559,33 @@
 #define CHERI_CLEAR_CAPHI_C31  (1 << (31 - 16))
 
 /*
- * Load symbol from capability table
+ * Helpers to load symbols from capability table
  */
-#define	CAPTABLE_LOAD(dst, sym)			\
-	clcbi dst, %captab20(sym)(CHERI_REG_GPC)
-#define	CAPCALL_LOAD(dst, sym)			\
-	clcbi dst, %capcall20(sym)(CHERI_REG_GPC)
+#define	GET_PCREL_CAPTABLE_PTR(dst, tmp)				\
+	.set push;							\
+	.set noat;							\
+	lui	tmp, %pcrel_hi(_CHERI_CAPABILITY_TABLE_-8);		\
+	daddiu	tmp, tmp, %pcrel_lo(_CHERI_CAPABILITY_TABLE_-4);	\
+	cgetpccincoffset dst, tmp;					\
+	.set pop
+
+#define	GET_ABS_CAPTABLE_PTR(dst, tmp)					\
+	ABSRELOC_LA(tmp, _CHERI_CAPABILITY_TABLE_);			\
+	cgetpccincoffset dst, tmp
```

Comment:
Should this be setaddr?

Reply:
Also is this even needed? Can't we use the pcrel version in `MipsCache`?

Reply:
Should definitely be setaddr. We need it for all uses in `VECTOR(xxx)` as that code is moved at a different address. `MipsCache` is one of those.

Reply:
Thanks, that makes sense.